src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: any[], key: keyof T): GroupsMap<object> {
```

Comment:
```suggestion
export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
```
lets make it work like this